### PR TITLE
fix(clouddriver/kubernetes) - Handle complex security group descripti…

### PR DESCRIPTION
**Pull request by johnrengelman**
…ons for kubernetes ingresses.

This addresses https://github.com/spinnaker/spinnaker/issues/2149

### CODE REVIEW

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/MortServiceSpec.groovy [hunk 1]
```diff
@@ -176,5 +183,114 @@ class MortServiceSpec extends Specification {
...
+      .create(MortService)
+
+    when:
+    def sg = mort.getSecurityGroup("account", "kubernetes", "sg1", "namespace")
+
+    then:
+    sg.description == "{\"account\":null,\"app\":\"sg1\"}"
+
+    1 * client.execute(_) >> new Response(
+      "http://localhost:999/securityGroups/account/kubernetes/namespace/sg1",
```
Comment (rossedman) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/MortServiceSpec.groovy]: You are missing a `9` here.

File: orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/MortServiceSpec.groovy [hunk 2]
```diff
@@ -176,5 +183,114 @@ class MortServiceSpec extends Specification {
...
+      .create(MortService)
+
+    when:
+    def sg = mort.getSecurityGroup("account", "openstack", "sg1", "region")
+
+    then:
+    sg.description == "simple description"
+
+    1 * client.execute(_) >> new Response(
+      "http://localhost:999/securityGroups/account/openstack/region/sg1",
```
Comment (rossedman) [inline orca-clouddriver/src/test/groovy/com/netflix/spinnaker/orca/clouddriver/MortServiceSpec.groovy]: And here as well

**Comment by rossedman** [comment]
Super awesome, I left a couple comments there are some typos in the PR for the `localhost` address.

**Comment by johnrengelman** [comment]
Fixed up those issues. They weren’t actually impacting the test in anyway
but it’s good for them to be realistic.

**Comment by rossedman** [comment]
@johnrengelman ah, i realized they were specs! sorry about that.

**Pull request merged**